R/checkers.R

```diff
@@ -0,0 +1,207 @@
+#' Check if `<data.frame>` defining age-stratified hospitalisation or death risk
+#' is correct
+#'
+#' @param x A `<data.frame>`.
+#'
+#' @return A `<data.frame>`, also called for error side-effects when input is
+#' invalid.
+#' @keywords internal
+.check_risk_df <- function(x, age_range) {
+  # check input
+  stopifnot(
+    "column names should be 'age_limit' & 'risk'" =
+      setequal(c("age_limit", "risk"), colnames(x)),
+    "minimum age of lowest age group should match lower age range" =
+      age_range[1] == min(x$age_limit),
+    "lower bound of oldest age group must be lower than highest age range" =
+      age_range[2] > max(x$age_limit),
+    "age limit or risk cannot be NA or NaN" =
+      !anyNA(x),
+    "risk should be between 0 and 1" =
+      min(x$risk) >= 0 && max(x$risk) <= 1,
+    "age limit in risk data frame must be unique" =
+      anyDuplicated(x$age_limit) == 0
+  )
+
+  # order risk df on age_limit
+  x <- x[order(x$age_limit), ]
+
+  # format risk data frame
+  age_range_ <- age_range[1]:age_range[2]
+  # findInterval inclusive/exclusive bound rules match age bracket
+  age_groups <- unname(split(age_range_, findInterval(age_range_, x$age_limit)))
+
+  # add and sort data frames cols
+  x$max_age <- vapply(age_groups, max, FUN.VALUE = numeric(1))
+  colnames(x) <- c("min_age", "risk", "max_age")
+  x <- x[, c("min_age", "max_age", "risk")]
+
+  # add informative row names
+  row.names(x) <- paste0(
+    "[", x$min_age, ",", (x$max_age + 1), ")"
+  )
+  # last age bracket has inclusive upper bound
+  row.names(x)[nrow(x)] <- paste0(
+    "[", x$min_age[nrow(x)], ",", x$max_age[nrow(x)], "]"
+  )
+
+  # return risk data frame
+  x
+}
+
+#' Check if `<data.frame>` defining age structure of population is correct
+#'
+#' @param x A `<data.frame>`.
+#'
+#' @return A `<data.frame>`, also called for error side-effects when input is
+#' invalid.
+#' @keywords internal
+.check_age_df <- function(x) {
+  # check input
+  stopifnot(
+    "column names should be 'age_range' & 'proportion'" =
+      setequal(c("age_range", "proportion"), colnames(x)),
```

Comment:
Suggest `x` should have three columns for `age_upper`, `age_lower`, and the proportion?

Reply:
I think what you're suggesting was the original implementation which was changed in #14. Please see this PR for details of the changes and the [first point in the design principles vignette](https://epiverse-trace.github.io/simulist/articles/design-principles.html#design-decisions) for reasoning. 

Let me know if there are compelling reasons to revert back the upper and lower age bounds.

Reply:
Okay thanks, understood - I think it's good to keep it as is then.